src/mapnik_image.cpp

```diff
@@ -1620,6 +1644,131 @@ void Image::EIO_AfterOpen(uv_work_t* req)
+        vertex_stl_adapter<svg_path_storage> stl_storage(marker_path->source());
+        svg_path_adapter svg_path(stl_storage);
+        svg_converter_type svg(svg_path, marker_path->attributes());
+        svg_parser p(svg);
+        if (fromFile)
+        {
+            p.parse(TOSTR(args[0]));
+        }
+        else
+        {
+            Local<Object> obj = args[0]->ToObject();
+            if (obj->IsNull() || obj->IsUndefined() || !node::Buffer::HasInstance(obj)) {
+                NanThrowTypeError("first argument is invalid, must be a Buffer");
+                return NanEscapeScope(NanUndefined());
+            }
+            std::string svg_buffer(node::Buffer::Data(obj),node::Buffer::Length(obj));
+            p.parse_from_string(svg_buffer);
+        }
+
+        double lox,loy,hix,hiy;
+        svg.bounding_rect(&lox, &loy, &hix, &hiy);
+        marker_path->set_bounding_box(lox,loy,hix,hiy);
+        marker_path->set_dimensions(svg.width(),svg.height());
+
+        using pixfmt = agg::pixfmt_rgba32_pre;
+        using renderer_base = agg::renderer_base<pixfmt>;
+        using renderer_solid = agg::renderer_scanline_aa_solid<renderer_base>;
+        agg::rasterizer_scanline_aa<> ras_ptr;
+        agg::scanline_u8 sl;
+
+        double opacity = 1;
+        int w = svg.width() * scale;
+        int h = svg.height() * scale;
+
+        mapnik::image_rgba8 im(w, h);
+        agg::rendering_buffer buf(im.bytes(), im.width(), im.height(), im.row_size());
+        pixfmt pixf(buf);
+        renderer_base renb(pixf);
+
+        mapnik::box2d<double> const& bbox = marker_path->bounding_box();
+        mapnik::coord<double,2> c = bbox.center();
+        // center the svg marker on '0,0'
+        agg::trans_affine mtx = agg::trans_affine_translation(-c.x,-c.y);
+        // Scale the image
+        mtx.scale(scale);
+        // render the marker at the center of the marker box
+        mtx.translate(0.5 * im.width(), 0.5 * im.height());
+
+        //mapnik::svg::vertex_stl_adapter<mapnik::svg::svg_path_storage> stl_storage(marker->get_data()->source());
+        //mapnik::svg::svg_path_adapter svg_path(stl_storage);
+        mapnik::svg::svg_renderer_agg<mapnik::svg::svg_path_adapter,
+            agg::pod_bvector<mapnik::svg::path_attributes>,
+            renderer_solid,
+            agg::pixfmt_rgba32_pre > svg_renderer_this(svg_path,
+                                                       marker_path->attributes());
+
+        svg_renderer_this.render(ras_ptr, sl, renb, mtx, opacity, bbox);
+        demultiply_alpha(im);
+
+        std::shared_ptr<mapnik::image_any> image_ptr = std::make_shared<mapnik::image_any>(im);
+        Image *im2 = new Image(image_ptr);
+        Handle<Value> ext = NanNew<External>(im2);
+        return NanEscapeScope(NanNew(constructor)->GetFunction()->NewInstance(1, &ext));
+    }
+    catch (std::exception const& ex)
+    {
+        NanThrowError(ex.what());
```

Comment:
Unit tests to cover this exception are probably required, I assume this would probably be used if an invalid buffer was passed?